src/main/java/bio/terra/common/kubernetes/KubePodListener.java

```diff
@@ -111,34 +111,16 @@ public void run() {
             consecutiveRetryCount = 0;
             retryWait = WATCH_INITIAL_WAIT;
 
-            String operation = item.type;
-            String podName =
-                (item.object.getMetadata() != null)
-                    ? item.object.getMetadata().getName()
-                    : org.apache.commons.lang3.StringUtils.EMPTY;
+            final String operation = item.type;
+            final String podName =
+                Optional.ofNullable(item.object.getMetadata())
+                    .map(V1ObjectMeta::getName)
+                    .orElse("");
             logger.info(String.format("%s : %s", operation, podName));
 
-            if (StringUtils.contains(podName, podNameFilter)) {
-              if (StringUtils.equals(operation, "ADDED")) {
-                logger.info("Added api pod: " + podName);
-                podMap.put(podName, PodState.RUNNING);
-              } else if (StringUtils.equals(operation, "DELETED")) {
-                try {
-                  logger.info("Attempting clean up of deleted stairway instance: " + podName);
-                  stairway.recoverStairway(podName);
-                  PodState deletedPodValue = podMap.get(podName);
-                  if (deletedPodValue == PodState.RUNNING) {
-                    logger.info("Deleted api pod: " + podName);
-                    podMap.put(podName, PodState.DELETED);
-                  }
-                } catch (DatabaseOperationException | StairwayExecutionException ex) {
-                  logger.error("Stairway recoverStairway failed to recover pod: " + podName, ex);
-                } catch (InterruptedException ex) {
-                  logger.info("KubePodListener interrupted - exiting", ex);
-                  exception = ex;
-                  return;
-                }
-              }
+            if (handlePodOperation(operation, podName)) {
```

Comment:
This is the most awkward bit. It returns a boolean letting you know to return early, but the name almost needs to be `handlePodOperation_isInterrupted`. 🤢

Reply:
That works all right, thanks.